### PR TITLE
[EMB-411][EmberOSF] Add CSRFToken to AJAX requests

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Add CSRFToken to authenticated AJAX requests

## Summary of Changes/Side Effects
- Modify tests
- Add token to config
- Use `Ember.assign` instead of soon deprecated `Ember.merge`

## Testing Notes

This will require full regression tests on registries, preprints
It also touches on `scheduled-banner` component so we could make sure it still works fine.

## Ticket

[EMB-411](https://openscience.atlassian.net/browse/EMB-411)

## Notes for Reviewer

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [x] testable and includes test(s)
- [x] changes described in `CHANGELOG.md`

### CODE REVIEW

File: addon/components/scheduled-banner/component.js [hunk 1]
```diff
@@ -54,17 +54,19 @@ export default Ember.Component.extend(Analytics, {
...
-                    defaultAltText: attrs.default_alt_text,
-                    mobileAltText: attrs.mobile_alt_text,
-                    defaultPhoto: links.default_photo,
-                    mobilePhoto: links.mobile_photo,
-                    startDate: attrs.start_date,
-                    link: attrs.link,
-                    name: attrs.name,
-                });
-                this.get('startDate') && this.set('hidden', false);
+                if (attrs.start_date) {
```
Comment (adlius) [inline addon/components/scheduled-banner/component.js]: Is this commit related to this ticket?

File: addon/mixins/infinity-custom.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import Ember from 'ember';
 import InfinityRoute from 'ember-infinity/mixins/route';
 
-const assign = Ember.assign || Ember.merge;
```
Comment (adlius) [inline addon/mixins/infinity-custom.js]: Maybe also note in the CHANGELOG that this PR remove usage of `Ember.merge` in favor of `Ember.assign`?

File: addon/adapters/osf-adapter.js [hunk 3]
```diff
@@ -21,12 +21,17 @@ import {
  * @uses GenericDataAdapterMixin
  */
 export default DS.JSONAPIAdapter.extend(GenericDataAdapterMixin, {
-    headers: {
-        ACCEPT: 'application/vnd.api+json; version=2.6'
-    },
+    currentUser: Ember.inject.service('current-user'),
```
Comment (adlius) [inline addon/adapters/osf-adapter.js]: Do we have to do `Ember.inject.service('current-user')` or is `Ember.inject.service()` sufficient? I think if the `Ember.inject.service()` is called with no arguments, it would automatically use the dasherized property name to look up the service.

File: addon/adapters/osf-adapter.js [hunk 4]
```diff
@@ -21,12 +21,17 @@ import {
...
+
     authorizer: config['ember-simple-auth'].authorizer,
     host: config.OSF.apiUrl,
     namespace: config.OSF.apiNamespace,
+
+    headers: Ember.computed('currentUser', function(){
+        const currentUser = this.get('currentUser');
+        return currentUser.ajaxHeaders();
+    }).volatile(),
```
Comment (adlius) [inline addon/adapters/osf-adapter.js]: :thumbsup:

File: package.json [hunk 5]
```diff
@@ -82,21 +82,21 @@
...
     "ember-cli-htmlbars-inline-precompile": "0.4.3",
+    "ember-cli-sass": "^6.2.0",
     "ember-cli-shims": "1.0.2",
-    "ember-concurrency": "^0.8.12",
-    "ember-feature-flags": "^5.0.0",
     "ember-collapsible-panel": "^2.1.1",
+    "ember-component-css": "^0.6.0",
     "ember-computed-style": "^0.2.0",
     "ember-concurrency": "^0.8.12",
+    "ember-cookies": "^0.3.1",
```
Comment (adlius) [inline package.json]: The addition of `ember-cookies` as an addon should be noted in the CHANGELOG.

File: addon/components/scheduled-banner/component.js (hunk 1 above)
Comment (fabmiz) [reply addon/components/scheduled-banner/component.js]: It was to fix some failing tests. I will use currentUser stub as well.
